timeutil/schedule.go

```diff
@@ -129,24 +129,38 @@ func (ws WeekSpan) String() string {
 	return ws.Start.String()
 }
 
+func findNthWeekDay(t time.Time, weekday time.Weekday, nthInMonth uint) time.Time {
+	// move to the beginning of the month
+	t = t.AddDate(0, 0, -t.Day()+1)
+
+	for nth := uint(0); t.Weekday() != weekday || nth != nthInMonth; {
+		t = t.Add(24 * time.Hour)
```

Comment:
Will this subtly break on leap year cases?

Reply:
AFAICT by looking at `time`, the `Time.Add()` only deals with seconds. Denomalizing happens only when doing Day(), Weekday() and so on. As long as that part is correct we should be fine.